### PR TITLE
develop Merge doxygen from 1.12 branch

### PR DESCRIPTION
**Pull request by byrnHDF**
Reference Manual Menus
Includes the fortran additions.
The Getting Started page and references to the Tools and Tutorials.
Java additions.
Fixes for refs in the UG

### CODE REVIEW

File: doxygen/examples/H5.format.2.0.html [hunk 1]
```diff
@@ -6610,7 +6610,7 @@ <h3><a name="SOHMTable">
 	<tr>
 	  <td><p>v2 B-tree Cutoff for index #N</p></td>
 	  <td>
-            <p>This is the cutoff value for the indexing of messages to
+            <p>This is is the cutoff value for the indexing of messages to
```
Comment (lrknox) [inline doxygen/examples/H5.format.2.0.html]:
```suggestion
            <p>This is the cutoff value for the indexing of messages to
```

File: fortran/src/H5Sff.F90 [hunk 2]
```diff
@@ -443,7 +443,7 @@ END SUBROUTINE h5sselect_none_f
 !!
 !! \brief Verifies that the selection is within the extent of the dataspace.
 !!
-!! \param space_id Identifier for the dataspace for whichselection is verified.
+!! \param space_id Identifier for the dataspace for whichselection is verified
```
Comment (lrknox) [inline fortran/src/H5Sff.F90]: whichselection missing a space?  I don't think it's a variable name.

File: java/src/hdf/hdf5lib/exceptions/HDF5LibraryException.java [hunk 3]
```diff
@@ -96,9 +106,11 @@ public HDF5LibraryException(String s)
     private native long _getMinorErrorNumber();
 
     /**
+     * @ingroup JERRLIB
+     *
      * Return a error message for the minor error number.
```
Comment (lrknox) [inline java/src/hdf/hdf5lib/exceptions/HDF5LibraryException.java]: Return an error message

**Review by lrknox — APPROVED** [review]

File: fortran/src/hdf5_fortrandll.def.in [hunk 4]
```diff
@@ -357,7 +357,7 @@ H5R_mp_H5RCREATE_REGION_F
 H5R_mp_H5RDEREFERENCE_OBJECT_F
 H5R_mp_H5RDEREFERENCE_REGION_F
 H5R_mp_H5RGET_REGION_REGION_F
-H5R_mp_H5RGET_OBJECT_TYPE_F
+H5R_mp_H5RGET_OBJECT_TYPE_OBJ_F
```
Comment (brtnfld) [inline fortran/src/hdf5_fortrandll.def.in:360]: This function was removed.

File: src/H5Dmodule.h [hunk 5]
```diff
@@ -28,7 +28,2961 @@
...
+ *
+ * \subsubsection subsubsec_dataset_transfer_drive File Drivers
+ * I/O is performed by the HDF5 virtual file layer. The file driver interface writes and reads blocks
+ * of data; each driver module implements the interface using different I/O mechanisms. The table
+ * below lists the file drivers currently supported. Note that the I/O mechanisms are separated from
+ * the pipeline processing: the pipeline and filter operations are identical no matter what data access
+ * mechanism is used.
+ *
+ * <table>
+ * <caption>I/O file drivers</caption>
```
Comment (jhendersonHDF) [inline src/H5Dmodule.h:1109]: Definitely will need to update this table in the future. Probably a candidate for pulling out into a separate file for maintenance reasons.

**Review by jhendersonHDF — APPROVED** [review]

File: src/H5Dmodule.h (hunk 5 above)
Comment (byrnHDF) [reply src/H5Dmodule.h:1109]: Create an issue for the idea!

File: fortran/src/hdf5_fortrandll.def.in (hunk 4 above)
Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: I think this gets handled by #2090

Comment (brtnfld) [reply fortran/src/hdf5_fortrandll.def.in:360]: ok, but it looks like this PR adds that function, removing H5R_mp_H5RGET_OBJECT_TYPE_F

**Review by brtnfld — APPROVED** [review]

File: fortran/src/hdf5_fortrandll.def.in (hunk 4 above)
Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: I'll check and fix

Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: 2090 is the same. The name should have _OBJ in it.

Comment (brtnfld) [reply fortran/src/hdf5_fortrandll.def.in:360]: This undoes #2057, the current develop is correct.

Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: But it won't compile on windows unless it is changed.

Comment (brtnfld) [reply fortran/src/hdf5_fortrandll.def.in:360]: correct, as this PR is now, it will not compile on windows.

Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: before this PR
:  75>hdf5_fortrandll.def : error LNK2001: unresolved external symbol H5R_mp_H5RGET_OBJECT_TYPE_F
-- COMMAND:  C:/autotest/hdf5trunk-StdShar-change-10vs16/build/ctest/Intel/bin/Release/f90_ex_compound.exe 
-- COMMAND Result: The system cannot find the file specified

after this PR:
Found 0 Errors
-- COMMAND:  C:/autotest/hdf5trunk-StdShar-change-10vs16/build/ctest/hdfbld/bin/Release/f90_ex_compound.exe 
-- COMMAND Result: 0
-- C:/autotest/hdf5trunk-StdShar-change-10vs16/build/ctest/hdfbld/bin/Release/f90_ex_compound.exe Passed

Comment (brtnfld) [reply fortran/src/hdf5_fortrandll.def.in:360]: Right, because PR #2057 changes are no longer in develop.

Comment (byrnHDF) [reply fortran/src/hdf5_fortrandll.def.in:360]: You changed that file in #2062

**Pull request merged**